lib/src/realm_object.dart

```diff
@@ -521,41 +559,41 @@ class DynamicRealmObject {
   /// in [Object].
   T get<T extends Object?>(String name) {
     _validatePropertyType<T>(name, RealmCollectionType.none);
-    return RealmObject.get<T>(_obj, name) as T;
+    return RealmObjectBase.get<T>(_obj, name) as T;
```

Comment:
Go to _propertyTypeMap and see that there is in item only for `RealmObject: RealmPropertyType.object`. Maybe should be added also for EmbeddedObject, it depends what is comming as a `name argument